### PR TITLE
[TF 2.3.0] [Intel MKL] remove duplicate registration for softmax bf16 op

### PR DESCRIPTION
**Pull request by nammbash**
Removing the duplicate registration of the softmax kernel from mkl_tmp_bf16_op.cc.

Google added the op registration here already
https://github.com/tensorflow/tensorflow/commit/cbac31d59ded64caa23a00481bb33d1104ab8822#diff-d960722b1c03feab5e375c2bfe409747R85

### CODE REVIEW

**Review by penpornk — APPROVED** [review]
Oops. My bad. I forgot to remove this when I added the `bfloat16` type to vanilla TF. Sorry about that and thank you for the fix!
